src/idl_gen_general.cpp

```diff
@@ -757,7 +757,9 @@ std::string GenKeyGetter(flatbuffers::FieldDef *key_field) {
       key_getter += ";";
   }
   else {
-    auto field_getter = data_buffer + GenGetter(key_field->value.type).substr(2) +
+    // offset is used to strip "bb" or "0!=bb" from GenGetter() result
+    auto offset = key_field->value.type.base_type == BASE_TYPE_BOOL ? 5 : 2;
```

Comment:
that's not a very clean way of doing things.. please instead pass a bool to not have `bb` or `0!=bb` not be generated in the first place.